Source/Core/sampleTerrain.js

```diff
@@ -1,11 +1,17 @@
 define([
         '../ThirdParty/when',
-        './Check'
+        './defined',
+        './Check',
+        './LRUCache'
     ], function(
         when,
-        Check) {
+        defined,
+        Check,
+        LRUCache) {
     'use strict';
 
+    var cache = new LRUCache(256, 10000);
```

Comment:
Maybe this was discussed and this is the best solution - if so please point me to it, but this is not how I originally suggested to implement this.

We generally want to centralize all the time-related cache logic so I suggested that we use the render loop to flush any caches just like we do for the shader cache.  Are we sure this `setInterval` approach is better?  I think explicit and consistent control of caches in our render loop (but still based on time stamp) would be more cohesive.

Reply:
There was some discussion here: https://github.com/AnalyticalGraphicsInc/cesium/pull/6284#discussion_r174890330

Since `sampleTerrain` doesn't have access to `frameState` some of the options were:
* Scene edits a global property in `sampleTerrain` that tells it to flush the cache.
* `sampleTerrain` uses `requestAnimationFrame` to hook into the render loop to flush the cache, making `sampleTerrain` self contained.
* `sampleTerrain` uses `setInterval` instead of `requestAnimationFrame`.

One benefit of the third approach (which I guess is opposite your original suggestion) is that `sampleTerrain` is not tied to the render loop at all, making it possible to use in Node.js or wherever else the Cesium engine isn't actually running.

Reply:
Ah I did not think about the standalone use case.  I'm still not sure that a standalone API would implicitly cache and flush like this (imagine if an OpenGL driver did thinks like this); the user would likely have more fine-grained control and would be able to pass in a cache policy or whatever.

Still not convinced that having random caches with random set intervals is a good design in Cesium, but proceed as you see fit.

Reply:
> Still not convinced that having random caches with random set intervals is a good design in Cesium

I think everyone agrees with this sentiment.  In my opinion what Cesium really needs is a shared centralized cache (just like we also need a centralized worker pool).